File: app/src/main/java/com/kirchhoff/example/githubclient/ui/auth/AuthContract.java
package com.kirchhoff.example.githubclient.ui.auth;

import android.support.annotation.Nullable;

import com.kirchhoff.example.githubclient.ui.BasePresenter;
import com.kirchhoff.example.githubclient.ui.BaseView;

public interface AuthContract {


    interface View extends BaseView {

        void openRepositoryScreen();

        void showLoginError();

        void showPasswordError();

    }

    interface Presenter extends BasePresenter {

        void auth(@Nullable String login, @Nullable String password);

    }
}

File: app/src/main/java/com/kirchhoff/example/githubclient/ui/auth/AuthContract.kt
package com.kirchhoff.example.githubclient.ui.auth

import com.kirchhoff.example.githubclient.ui.BasePresenter
import com.kirchhoff.example.githubclient.ui.BaseView

/**
 * @author Kirchhoff-
 */
interface AuthContract {

    interface View : BaseView {

        fun openRepositoryScreen()

        fun showLoginError()

        fun showPasswordError()
    }

    interface Presenter : BasePresenter {

        fun auth(login: String, password: String)
    }
}